hooks/pre-receive

```diff
@@ -153,8 +154,11 @@ if (isset($closedBranches[$repo_name])) {
 }
 
 $restricted = [];
+if (strpos($repo_name, 'pecl/') !== 0  && in_array('refs/heads/master', $pi->getBranches())) {
+	$restricted[] = 'refs/heads/master';
+}
 if (isset($restrictedBranches[$repo_name])) {
-    $restricted = array_filter($restrictedBranches[$repo_name],
+    $restricted += array_filter($restrictedBranches[$repo_name],
```

Comment:
The `+` operator will give the wrong behaviour when `$restricted` is not empty, the first filtered value will not be present in the resulting array.  You'll want to use `array_merge()` instead or append the master ref later on.